src/modules/googlesheets.ts

```diff
@@ -0,0 +1,149 @@
+import axios from 'axios';
+import _ from 'lodash';
+import { AccessInfoBigQueryModel } from '../generated/tdr';
+import { showNotification } from './notifications';
+
+export type SpreadsheetInfo = {
+  spreadsheetId: string;
+  spreadsheetUrl: string;
+};
+
+export type SheetInfo = {
+  sheetId: string;
+  title: string;
+};
+
+export const createSheet: any = async (sheetName: string, token: string) => {
+  // Using google drive api endpoint rather than google sheets api
+  // in order to facilitate file delete on error
+  const url = '/drive/v3/files';
+  return axios
+    .post(
+      url,
+      { mimeType: 'application/vnd.google-apps.spreadsheet', name: sheetName },
+      {
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`,
+        },
+      },
+    )
+    .then((response) => ({
+      spreadsheetId: response.data.id,
+      spreadsheetUrl: `https://docs.google.com/spreadsheets/d/${response.data.id}`,
+    }))
+    .catch((err) => {
+      showNotification(err);
+    });
+};
+
+// Adding data sources for each table one by one
+// In theory, could do a batch request for all data sources
+// But ran into errors around concurrent requests from the sheets api
+export const addBQSources: any = async (
+  spreadsheetId: string,
+  bigQueryAccessInfo: AccessInfoBigQueryModel,
+  token: string,
+) => {
+  const url = `/googlesheets/v4/spreadsheets/${spreadsheetId}:batchUpdate`;
+  const sheetInfo: Array<SheetInfo> = [];
+  let index = 0;
+  for (const table of bigQueryAccessInfo.tables) {
+    const requests: object[] = [
+      {
+        addDataSource: {
+          dataSource: {
+            spec: {
+              bigQuery: {
+                projectId: bigQueryAccessInfo.projectId,
+                querySpec: {
+                  rawQuery: table.sampleQuery,
```

Comment:
I think this is fine for this PR but longer term, I think we may want to change either the api or the query here to explicitly use columns so that we have consistent column ordering

Reply:
Ah, great point. I would lean towards updating the sample query, but I suppose that could get a bit unmanageable for tables with a lot of columns. I'll put in a ticket.

Reply:
https://broadworkbench.atlassian.net/browse/DR-2604